packages/common/src/providers/httpProvider.ts

```diff
@@ -0,0 +1,39 @@
+import Provider, { JsonRpcResponse } from './provider';
+import fetch from 'cross-fetch';
+
+export class HttpProvider implements Provider {
+    request: (
+        method: string,
+        params?: Record<string, unknown>
+    ) => Promise<JsonRpcResponse>;
+    nextId = 0;
+
+    constructor(url: string) {
+        this.request = async function (
+            method: string,
+            params?: Record<string, unknown>
+        ) {
+            const request = {
```

Comment:
We have a nice type for the response, so we should also have one for the request.

Reply:
I tried creating one.

Reply:
But you didn't use it here? I would expect `request` to have an explicit type :smile: